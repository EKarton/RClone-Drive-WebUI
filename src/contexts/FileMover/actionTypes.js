const actionTypes = Object.freeze({
  REQUEST_FILE_MOVE: 'REQUEST_FILE_RENAME',
  COMPLETE_FILE_MOVE: 'COMPLETE_FILE_RENAME',
  CANCEL_FILE_MOVE: 'CANCEL_FILE_RENAME',
});

export default actionTypes;
