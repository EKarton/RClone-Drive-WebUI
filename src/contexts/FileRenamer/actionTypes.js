const actionTypes = Object.freeze({
  REQUEST_FILE_RENAME: 'REQUEST_FILE_RENAME',
  COMPLETE_FILE_RENAME: 'COMPLETE_FILE_RENAME',
  CANCEL_FILE_RENAME: 'CANCEL_FILE_RENAME',
});

export default actionTypes;
