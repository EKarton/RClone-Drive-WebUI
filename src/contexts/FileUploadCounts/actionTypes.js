const actionTypes = Object.freeze({
  UPDATE_STATUS: 'UPDATE_STATUS',
  ADD_STATUS: 'ADD_STATUS',
});

export default actionTypes;
