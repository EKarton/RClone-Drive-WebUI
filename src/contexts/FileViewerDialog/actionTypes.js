const actionTypes = Object.freeze({
  SHOW_DIALOG: 'SHOW_DIALOG',
  HIDE_DIALOG: 'HIDE_DIALOG',
  SET_FILE_INFO: 'SET_FILE_INFO',
});

export default actionTypes;
