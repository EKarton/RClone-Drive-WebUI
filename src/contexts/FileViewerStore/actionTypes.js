const actionTypes = Object.freeze({
  SHOW_DIALOG: 'SHOW_DIALOG',
  HIDE_DIALOG: 'HIDE_DIALOG',
  SET_DIALOG_VISIBILITY: 'SET_DIALOG_VISIBILITY',
  SET_FILE_INFO: 'SET_FILE_INFO',
});

export default actionTypes;
