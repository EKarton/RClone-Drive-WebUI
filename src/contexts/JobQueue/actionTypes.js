export const ActionTypes = Object.freeze({
  SET_JOBS: 'SET_JOBS',
  ADD_JOB: 'ADD_JOB',
  ADD_JOBS: 'ADD_JOBS',
  REMOVE_JOB: 'REMOVE_JOB',
  ADD_STATUS_COUNT: 'ADD_STATUS_COUNT',
  UPDATE_STATUS: 'UPDATE_STATUS',
});
