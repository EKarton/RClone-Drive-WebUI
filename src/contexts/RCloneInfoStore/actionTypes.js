const actionTypes = {
  SET_ENDPOINT: 'SET_ENDPOINT',
  SET_USERNAME: 'SET_USERNAME',
  SET_PASSWORD: 'SET_PASSWORD',
};

export default actionTypes;
