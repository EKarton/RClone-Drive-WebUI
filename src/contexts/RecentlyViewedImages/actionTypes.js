const actionTypes = {
  ADD_IMAGE: 'ADD_IMAGE',
};

export default actionTypes;
