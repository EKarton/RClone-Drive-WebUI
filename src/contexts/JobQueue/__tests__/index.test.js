import { useContext } from 'react';
import { JobStatus } from 'utils/constants';
import { render, userEvent, screen } from 'test-utils/react';
import { ActionTypes } from '../actionTypes';
import { JobQueueContext, JobQueueProvider } from '../index';

describe('JobQueueProvider', () => {
  it('should render initial state given mock component renders state', () => {
    const MockComponent = () => {
      const { state } = useContext(JobQueueContext);

      return <div>{JSON.stringify(state)}</div>;
    };

    const { baseElement } = render(
      <JobQueueProvider>
        <MockComponent />
      </JobQueueProvider>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should update state given mock component dispatches event', async () => {
    const MockComponent = () => {
      const { state, dispatch } = useContext(JobQueueContext);

      const handleClick = () => {
        dispatch({ type: ActionTypes.ADD_JOB, payload: { jobId: '123' } });
        dispatch({ type: ActionTypes.ADD_STATUS_COUNT, payload: JobStatus.ONGOING });
      };

      return (
        <>
          <div>{JSON.stringify(state)}</div>
          <div>Num jobs: {state.jobs.length}</div>
          <button onClick={handleClick}>Update state</button>
        </>
      );
    };

    const { baseElement } = render(
      <JobQueueProvider>
        <MockComponent />
      </JobQueueProvider>
    );

    userEvent.click(screen.getByText('Update state'));

    await screen.findByText('Num jobs: 1');
    expect(baseElement).toMatchSnapshot();
  });
});
