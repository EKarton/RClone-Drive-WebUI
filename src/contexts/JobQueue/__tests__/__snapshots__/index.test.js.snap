// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobQueueProvider should render initial state given mock component renders state 1`] = `
<body>
  <div>
    <div>
      {"jobs":[],"statusCounts":{"numOngoing":0,"numSuccessful":0,"numFailed":0}}
    </div>
  </div>
</body>
`;

exports[`JobQueueProvider should update state given mock component dispatches event 1`] = `
<body>
  <div>
    <div>
      {"jobs":[{"jobId":"123"}],"statusCounts":{"numOngoing":1,"numSuccessful":0,"numFailed":0}}
    </div>
    <div>
      Num jobs: 
      1
    </div>
    <button>
      Update state
    </button>
  </div>
</body>
`;
